VideoLocker/src/main/java/org/edx/mobile/http/NewVersionBroadcastInterceptor.java

```diff
@@ -0,0 +1,75 @@
+package org.edx.mobile.http;
+
+import org.apache.commons.lang3.time.DateFormatUtils;
+import org.edx.mobile.BuildConfig;
+import org.edx.mobile.event.NewVersionAvailableEvent;
+import org.edx.mobile.third_party.versioning.ArtifactVersion;
+import org.edx.mobile.third_party.versioning.DefaultArtifactVersion;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.util.Date;
+
+import okhttp3.Interceptor;
+import okhttp3.Response;
+
+/**
+ * An OkHttp interceptor that checks for information about app
+ * updates in the response headers, and broadcasts them on the event
+ * bus if found.
+ */
+public class NewVersionBroadcastInterceptor implements Interceptor {
+    /**
+     * Header field name for the latest version number of the app
+     * that is available in the app stores.
+     */
+    private static final String HEADER_APP_LATEST_VERSION =
+            "EDX-APP-LATEST-VERSION";
+    /**
+     * Header field name for the last date up to which the API used
+     * by the current version of the app will be supported and
+     * maintained.
+     */
+    private static final String HEADER_APP_VERSION_LAST_SUPPORTED_DATE =
+            "EDX-APP-VERSION-LAST-SUPPORTED-DATE";
+
+    @Override
+    public Response intercept(final Chain chain) throws IOException {
+        final Response response = chain.proceed(chain.request());
+
+        final String appLatestVersionString = response.header(HEADER_APP_LATEST_VERSION);
+        final ArtifactVersion appLatestVersion = appLatestVersionString == null ?
+                null : new DefaultArtifactVersion(appLatestVersionString);
+
+        final String lastSupportedDateString =
+                response.header(HEADER_APP_VERSION_LAST_SUPPORTED_DATE);
+        Date lastSupportedDate = null;
+        if (lastSupportedDateString != null) {
+            /* We're using Apache Commons Lang's FastDateFormat here, because it
+             * has improved support for ISO 8601, including UTC offsets and the
+             * "Z" notation.
+             * TODO: Implement this all over the app instead of the existing
+             * SimpleDateFormat.
```

Comment:
So, it looks like the `XXX` pattern was only introduced in Java 7, which is only supported in KitKat+. Fortunately, however, I found out that Gson 2.5 imported an ISO 8601 parser from Jackson, so I upgraded Gson to 2.7 and changed our `DateUtils` class to call through to it instead in commit 2e7fb35587e810b98cd46bb75936f957384667d3. I also changed our date type adapter to use that, since Gson still doesn't include a dedicated type adapter for ISO 8601 dates in the main library (it only falls back to it on deserialization, but formats to a specified or default date/time pattern using the device's local time zone when serializing 😞).